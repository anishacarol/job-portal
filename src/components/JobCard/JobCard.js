import React from "react";
import { Link } from "react-router-dom";
import * as Styled from "./JobCard.styles";

const JobCard = () => {
  return (
    <Styled.JobCard>
      <Styled.Company>SmartReqruiters</Styled.Company>
      <Styled.JobTitle>Frontend Engineer</Styled.JobTitle>
      <Styled.Information>
        <span>Berlin</span>
        <span>full-time</span>
      </Styled.Information>
      <Styled.LinkHolder>
        <Link to="job-list">Read more..</Link>
      </Styled.LinkHolder>
    </Styled.JobCard>
  );
};

export default JobCard;
