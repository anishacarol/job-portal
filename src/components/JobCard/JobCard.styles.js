import styled from "styled-components";

export const JobCard = styled.div`
  min-height: 90px;
  background: #fff;
  padding: 15px 50px;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  margin-top: 15px;

  &:hover {
    box-shadow: rgba(0, 0, 0, 0.2) 0px 8px 24px;
  }

  @media (max-width: 768px) {
    flex-direction: column;
  }
`;

export const Company = styled.span`
  display: block;
  color: rgb(155, 155, 155);
  margin-bottom: 14px;
`;

export const JobTitle = styled.span`
  display: block;
  color: rgb(28, 28, 28);
  font-weight: 600;
  margin-bottom: 16px;
  font-size: 18px;
`;

export const JobInfo = styled.div`
  display: flex;
  flex-direction: column;
`;

export const Information = styled.div`
  display: flex;

  @media (max-width: 768px) {
    flex-direction: column;

    span {
      margin-bottom: 18px;
    }
  }

  span {
    font-size: 14px;
    font-weight: 400;
    display: flex;
    align-items: center;
    margin-right: 30px;

    svg {
      margin-right: 10px;
    }
  }
`;

export const LinkHolder = styled.div`
  align-self: center;
  margin-top: 50px;

  @media (max-width: 768px) {
    display: flex;
    width: 100%;
    margin-top: 0;
    text-align: center;

    a {
      width: 100%;
    }
  }
`;
