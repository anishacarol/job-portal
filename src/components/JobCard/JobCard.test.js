import { render } from "@testing-library/react";
import React from "react";
import { MemoryRouter } from "react-router-dom";
import JobCard from "./JobCard";

const mockData = {
  position: "frontend",
  companyName: "job gmbh",
  country: "berlin",
  typeOfEmployment: "full_time"
};

describe("<JobCard/>", () => {
  it("renders JobCard component properly", () => {
    const { asFragment, getByText } = render(
      <MemoryRouter>
        <JobCard {...mockData} />
      </MemoryRouter>
    );
    const position = getByText(/frontend/i);
    const companyName = getByText(/job gmbh/i);
    const country = getByText(/berlin/i);
    const typeOfEmployment = getByText(/full_time/i);

    expect(position).toBeInTheDocument();
    expect(companyName).toBeInTheDocument();
    expect(country).toBeInTheDocument();
    expect(typeOfEmployment).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });
});
