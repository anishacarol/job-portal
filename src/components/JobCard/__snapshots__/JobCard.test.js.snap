// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<JobCard/> renders JobCard component properly 1`] = `
<DocumentFragment>
  <div
    class="sc-AxirZ ffkYTh"
  >
    <div
      class="sc-AxhUy gfXgDx"
    >
      <span
        class="sc-AxiKw bAusMf"
      >
        job gmbh
      </span>
      <span
        class="sc-AxhCb hHvETg"
      >
        frontend
      </span>
      <div
        class="sc-AxgMl keSmOc"
      >
        <span>
          <svg
            height="20"
            version="1.1"
            viewBox="0 0 32 32"
            width="20"
            xml:space="preserve"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              clip-rule="evenodd"
              fill="#333"
              fill-rule="evenodd"
            >
              <path
                d="M16.002 17.746c3.309 0 6-2.692 6-6s-2.691-6-6-6-6 2.691-6 6 2.691 6 6 6zm0-11c2.758 0 5 2.242 5 5s-2.242 5-5 5-5-2.242-5-5 2.242-5 5-5z"
              />
              <path
                d="M16 0C9.382 0 4 5.316 4 12.001c0 7 6.001 14.161 10.376 19.194.016.02.718.805 1.586.805h.077c.867 0 1.57-.785 1.586-.805 4.377-5.033 10.377-12.193 10.377-19.194A11.971 11.971 0 0016 0zm.117 29.883c-.021.02-.082.064-.135.098-.01-.027-.084-.086-.129-.133C12.188 25.631 6 18.514 6 12.001 6 6.487 10.487 2 16 2c5.516 0 10.002 4.487 10.002 10.002 0 6.512-6.188 13.629-9.885 17.881z"
              />
            </g>
          </svg>
          undefined , undefined
        </span>
        <span>
          <svg
            height="20"
            viewBox="0 0 20 20"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              fill="none"
              fill-rule="evenodd"
              stroke="none"
              stroke-width="1"
            >
              <g
                fill="#000"
                fill-opacity="0.9"
                transform="translate(-44)"
              >
                <g
                  transform="translate(44)"
                >
                  <path
                    d="M10 0C4.5 0 0 4.5 0 10s4.5 10 10 10 10-4.5 10-10S15.5 0 10 0zm0 18c-4.4 0-8-3.6-8-8s3.6-8 8-8 8 3.6 8 8-3.6 8-8 8zm.5-13H9v6l5.2 3.2.8-1.3-4.5-2.7V5z"
                  />
                </g>
              </g>
            </g>
          </svg>
        </span>
      </div>
    </div>
    <div
      class="sc-AxheI kpmTr"
    >
      <a
        class="sc-AxjAm fziQwd"
        href="/job-details/undefined"
      >
        Read more &gt;
      </a>
    </div>
  </div>
</DocumentFragment>
`;
